autoload/airline/extensions/branch.vim

```diff
@@ -206,13 +251,15 @@ function! s:check_in_path()
 endfunction
 
 function! s:reset_untracked_cache(shellcmdpost)
-  if a:shellcmdpost
-    " function called after executing a shell command,
-    " only clear cache, if there was no error, else the
-    " system() command from get_git_untracked() would
-    " overwrite the v:shell_error status
-    if v:shell_error
-      return
+  if !s:has_async
+    if a:shellcmdpost
+      " function called after executing a shell command,
+      " only clear cache, if there was no error, else the
```

Comment:
The current comment has small grammar problems. Also, It might be improved by explaining `!s:has_async` condition. Consider this comment

> Clear cache only if there was no error or the script uses an asynchronous interface. Otherwise, cache clearing would overwrite v:shell_error with a system() call inside get_*_untracked.